pyproject.toml

```diff
@@ -12,27 +15,45 @@ python = ">=3.9"
 patchdiff = "~0.3.4"
 
 [tool.poetry.group.dev.dependencies]
-black = "*"
-flake8 = "*"
-flake8-black = "*"
-flake8-import-order = "*"
-flake8-print = "*"
+ruff = "*"
 pre-commit = "*"
 pytest = "*"
 pytest-benchmark = "*"
 pytest-cov = "*"
 pytest-timeout = "*"
 twine = "*"
-urllib3 = { version = "*", python = "<4"}
+urllib3 = { version = "*", python = "<4" }
 
 [tool.poetry.group.qt]
 optional = true
 
 [tool.poetry.group.qt.dependencies]
-PySide6 = { version = ">=6.6", python = "<3.13"}
+PySide6 = { version = ">=6.6", python = "<3.13" }
 pytest-qt = "*"
 pytest-xvfb = "*"
 
+[tool.ruff.lint]
+select = [
+    "E4",
+    "E7",
+    "E9",
```

Comment:
I see you left out E5 (line too long) which I agree is super annoying. But does the formatter still apply max line length?

Reply:
I picked the default set of rules from ruff and expanded on it a bit. E5 just wasn't in there.

> While the [line-too-long](https://docs.astral.sh/rules/line-too-long/) (E501) rule can be used alongside the formatter, the formatter only makes a best-effort attempt to wrap lines at the configured [line-length](https://docs.astral.sh/settings/#line-length). As such, formatted code may exceed the line length, leading to [line-too-long](https://docs.astral.sh/rules/line-too-long/) (E501) errors. 
(https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules)

So the formatter tries its best to keep line-length in check (default is 88, like black).

On the one hand, it might be annoying to have E501, but long lines themselves can be annoying as well... So I'd rather have shorter lines :)